app/graph.py

```diff
@@ -471,23 +489,119 @@ def update(self):
+                                                    'switch_name': switch_name,
+                                                    'switch_interface': switch_interface,
+                                                    'node_iface': iface_name
+                                                })
+
+                                        for pod in pods:
+                                            node_iface = pod.get("localIface")
+                                            if "sriov" in i["spec"]["primaryCni"]:
+                                                node_iface = "VF-" + node_iface
+                                            pod_name = pod.get("podRef")["name"]
+                                            network_ref = i["spec"].get("networkRef")
+                                            node_network = network_ref["name"]
+                                            if "sriov" in i["spec"]["primaryCni"]:
+                                                self.sriov = True
+                                                self.topology['nodes'][nodeName]['node_pod_sec_iface_conn'].append({
+                                                    'node_iface': node_iface,
+                                                    'pod_name': pod_name,
+                                                    'node_network': node_network,
+                                                    'pod_iface': self.topology['nodes'][nodeName]['pods'][pod_name]['other_ifaces'].get(node_network, "")
+                                                })
+                                            else:
+                                                self.macvlan = True
+                                                self.topology['nodes'][nodeName]['node_pod_ter_iface_conn'].append({
+                                                    'node_iface': node_iface,
+                                                    'pod_name': pod_name,
+                                                    'node_network': node_network,
+                                                    'pod_iface': self.topology['nodes'][nodeName]['pods'][pod_name]['other_ifaces'].get(node_network, "")
+                                                })
+
+                        self.topology['nodes'][nodeName]['node_leaf_all_iface_conn'].extend(self.topology['nodes'][nodeName]['node_leaf_sec_iface_conn'])
+                        self.topology['nodes'][nodeName]['node_leaf_all_iface_conn'].extend(self.topology['nodes'][nodeName]['node_leaf_ter_iface_conn'])
+
+                    except Exception as e:
+                        logger.error(f"Error processing node: {nodeName}. Error: {str(e)}")
+
+        except Exception as e:
+            if e.status == 404:
+                logger.info(f"CRD nodefabricnetworkattachments not detected, SR-IOV/MACVLAN topology support is disabled")
```

Comment:
Was it meant to be an f string? Or did you guys plan to add a variable to it?

Reply:
noted